### DISCUSSION TITLE
node package manager

### DISCUSSION THREAD

**Discussion by nicks-passiert** [topgrade-rs/topgrade#192] [Q&A]
Hi! I don't understand how I could disable npm and pnpm to run only yarn. Is there a solution?

**Comment by s34m** [comment]
See #182 and #170

**Reply by nicks-passiert** [reply]
Thank you

**Reply by nicks-passiert** [reply]
Hi @DottoDev maybe I'm asking a stupid question but from your linked issue/pull request I'm not getting a clue how to disable npm and pnpm. Could you help me with this?

**Reply by s34m** [reply]
This is one valid option by @Nieto2018 

> Hi, 
> 
> 
> 
> I use [nvm](https://github.com/nvm-sh/nvm) and to solve it I done the following.
> 
> 
> 
> - Run `topgrade --only node -y --verbose`
> 
> 
> 
> Logs showed that topgrade runs the command "/home/<user_dir>/.nvm/versions/node/v19.1.0/bin/pnpm root -g", which fails.
> 
> 
> 
> I runned the same command manually to check the error:
> 
> 
> 
> ```
> 
> $ home/<user_dir>/.nvm/versions/node/v19.1.0/bin/pnpm root -g
> 
>  ERROR  Unable to find the global bin directory
> 
> Run "pnpm setup" to create it automatically, or set the global-bin-dir setting, or the PNPM_HOME env variable. The global bin directory should be in the PATH.
> 
> ```
> 
> 
> 
> and next:
> 
> 
> 
> ```
> 
> $ pnpm setup
> 
> Appended new lines to /home/<user_dir>/.zshrc
> 
> 
> 
> Next configuration changes were made:
> 
> export PNPM_HOME="/home/<user_dir>/.local/share/pnpm"
> 
> export PATH="$PNPM_HOME:$PATH"
> 
> 
> 
> To start using pnpm, run:
> 
> source /home/<user_dir>/.zshrc
> 
> ```
> 
> if `pnpm setup` fails, try with `/home/<user_dir>/.nvm/versions/node/v19.1.0/bin/pnpm setup`
> 
> 
> 
> Take care because your **pnpm** path could be different.
> 
> 
> 
> It works for me, but if the solutions doesn't work you can skip this step with `topgrade --disable node -y`

**Reply by s34m** [reply]
But this just fixes pnpm, there is no real way to disable one of them because of a weird bug Noone really can explain.

**Reply by nicks-passiert** [reply]
I understand, then it's easier for me to let pnpm not retry...
